api/v1alpha1/clusterconfig_types.go

```diff
@@ -237,52 +241,78 @@ func (ExtraAPIServerCertSANs) VariableSchema() clusterv1.VariableSchema {
 	}
 }
 
-type ImageRegistries struct {
+type ImageCredentials struct {
+	// The Secret containing the registry credentials and CA certificate
+	// The Secret should have keys 'username', 'password' and 'ca.crt'
+	// This credentials Secret is not required for some registries, e.g. ECR.
 	// +optional
-	ImageRegistryCredentials ImageRegistryCredentials `json:"credentials,omitempty"`
+	SecretRef *corev1.ObjectReference `json:"secretRef,omitempty"`
```

Comment:
Do we want to be able to reference secrets from other namespaces? That has the potential of leaking registry credentials. I think this should be a `LocalObjectReference`.
```suggestion
	SecretRef *corev1.LocalObjectReference `json:"secretRef,omitempty"`
```

Reply:
+1

Reply:
TIL: https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/local-object-reference/

Reply:
The [current implementation](https://github.com/d2iq-labs/capi-runtime-extensions/blob/main/pkg/handlers/generic/mutation/imageregistries/credentials/inject.go#L319) allows secret from different namespace. If we have to change this, it touches many files and tests. Is it okay if we can fix this in a separate PR?

Reply:
I created https://github.com/d2iq-labs/capi-runtime-extensions/issues/322.

Reply:
Thank you